ably-ios/ARTPayload.m

```diff
@@ -63,6 +65,44 @@ + (instancetype)payloadWithPayload:(id)payload encoding:(NSString *)encoding {
         [[ARTCipherPayloadEncoder alloc] initWithCipherParams:cipherParams]]];
 }
 
++(id<ARTPayloadEncoder>) createEncoder:(NSString *) name key:(NSData *) key iv:(NSData *) iv {
+    if([name isEqualToString:@"json"]) {
+        return [ARTJsonPayloadEncoder instance];
+    }
+    else if([name isEqualToString:@"base64"]) {
+        return [ARTBase64PayloadEncoder instance];
+    }
+    else if([name isEqualToString:@"utf-8"]) {
+        return [ARTUtf8PayloadEncoder instance];
+    }
+    else if([name isEqualToString:@"cipher+aes-256-cbc"] || [name isEqualToString:@"cipher+aes-128-cbc"]){
+        
+        ARTIvParameterSpec * ivSpec = [[ARTIvParameterSpec alloc] initWithIv:iv];
+        ARTSecretKeySpec * keySpec = [[ARTSecretKeySpec alloc] initWithKey:key algorithm:@"aes"];
+        ARTCipherParams * params =[[ARTCipherParams alloc] initWithAlgorithm:@"aes" keySpec:keySpec ivSpec:ivSpec];
+
+        return [[ARTCipherPayloadEncoder alloc] initWithCipherParams:params];
```

Comment:
I'm not sure I follow how this code will work with different keylengths as from what I can tell, it key length part is ignored?